### ISSUE TITLE
Error generate graphql types when following tutorial, trying to create a cell

### ISSUE THREAD

**Issue by MrEmanuel** [redwoodjs/redwood#5708]
Hi! First off, awesome project! 

I'm following the tutorial and running in to some issues that I can't solve. 
I'm trying to create the cell with: 
`yarn rw g cell Articles` 

This gives me the following error message: 
```
(node:930) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
  ✔ Generating cell files...
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.mock.ts`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.test.tsx`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.stories.tsx`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.tsx`
  ⠹ Generating types ...

Error: Could not generate GraphQL type definitions (web)
AggregateError: GraphQL Document Validation failed with 1 errors;
  Error 0: GraphQLDocumentError: Cannot query field "articles" on type "Query".
    at /home/emanuel/source/redwoodblog/web/src/components/ArticlesCell/ArticlesCell.tsx:2:3
    at Object.checkValidationErrors (/home/emanuel/source/redwoodblog/node_modules/@graphql-tools/utils/index.js:1277:15)
  ✔ Generating cell files...
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.mock.ts`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.test.tsx`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.stories.tsx`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.tsx`
  ✔ Generating types ...

```

Earlier in the documentation it says: 
> If you head back to VSCode at some point and get a notice in one of the generated Post cells about Cannot query "posts" on type "Query" don't worry: we've seen this from time to time on some systems. There are two easy fixes:

> Run yarn rw g types in a terminal
Reload the GraphQL engine in VSCode: open the Command Palette (Cmd+Shift+P for Mac, Ctrl+Shift+P for Windows) and find "VSCode GraphQL: Manual Restart"

When I follow the instructions and generate the types again with `yarn rw g types` I get this error: 
```
(node:1170) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

Generating...

(node:1218) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)

Error: Could not generate GraphQL type definitions (web)
AggregateError: GraphQL Document Validation failed with 1 errors;
  Error 0: GraphQLDocumentError: Cannot query field "articles" on type "Query".
    at /home/emanuel/source/redwoodblog/web/src/components/ArticlesCell/ArticlesCell.tsx:2:3
    at Object.checkValidationErrors (/home/emanuel/source/redwoodblog/node_modules/@graphql-tools/utils/index.js:1277:15)
    at codegen (/home/emanuel/source/redwoodblog/node_modules/@graphql-codegen/core/index.js:175:15)
    at async runCodegenGraphQL (/home/emanuel/source/redwoodblog/node_modules/@redwoodjs/internal/dist/generate/graphqlCodeGen.js:119:18)
    at async generateTypeDefGraphQLWeb (/home/emanuel/source/redwoodblog/node_modules/@redwoodjs/internal/dist/generate/graphqlCodeGen.js:89:12)
    at async generateTypeDefs (/home/emanuel/source/redwoodblog/node_modules/@redwoodjs/internal/dist/generate/typeDefinitions.js:63:18)
    at async generate (/home/emanuel/source/redwoodblog/node_modules/@redwoodjs/internal/dist/generate/generate.js:26:25)
    at async run (/home/emanuel/source/redwoodblog/node_modules/@redwoodjs/internal/dist/generate/generate.js:37:17)

- .redwood/schema.graphql
- .redwood/types/mirror/api/src/directives/requireAuth/index.d.ts
- .redwood/types/mirror/api/src/directives/skipAuth/index.d.ts
- .redwood/types/mirror/api/src/services/posts/index.d.ts
- .redwood/types/mirror/web/src/layouts/BlogLayout/index.d.ts
- .redwood/types/mirror/web/src/layouts/PostsLayout/index.d.ts
- .redwood/types/mirror/web/src/pages/AboutPage/index.d.ts
- .redwood/types/mirror/web/src/pages/FatalErrorPage/index.d.ts
- .redwood/types/mirror/web/src/pages/HomePage/index.d.ts
- .redwood/types/mirror/web/src/pages/NotFoundPage/index.d.ts
- .redwood/types/mirror/web/src/components/Post/NewPost/index.d.ts
- .redwood/types/mirror/web/src/components/Post/Post/index.d.ts
- .redwood/types/mirror/web/src/components/Post/PostForm/index.d.ts
- .redwood/types/mirror/web/src/components/Post/Posts/index.d.ts
- .redwood/types/mirror/web/src/pages/Post/EditPostPage/index.d.ts
- .redwood/types/mirror/web/src/pages/Post/NewPostPage/index.d.ts
- .redwood/types/mirror/web/src/pages/Post/PostPage/index.d.ts
- .redwood/types/mirror/web/src/pages/Post/PostsPage/index.d.ts
- .redwood/types/mirror/web/src/components/ArticlesCell/index.d.ts
- .redwood/types/mirror/web/src/components/Post/EditPostCell/index.d.ts
- .redwood/types/mirror/web/src/components/Post/PostCell/index.d.ts
- .redwood/types/mirror/web/src/components/Post/PostsCell/index.d.ts
- .redwood/types/includes/web-routesPages.d.ts
- .redwood/types/includes/all-currentUser.d.ts
- .redwood/types/includes/web-routerRoutes.d.ts
- .redwood/types/includes/api-globImports.d.ts
- .redwood/types/includes/api-globalContext.d.ts
- .redwood/types/includes/api-scenarios.d.ts
- .redwood/types/includes/api-test-globals.d.ts
- .redwood/types/includes/web-test-globals.d.ts
- api/types/graphql.d.ts

... and done.
```

What I've tried: 

- Restarted the vscode graphql server multiple times.
- Restarted vscode
- Run and restart the redwood app multiple times (runs fine, because the article isn't implemented anywhere yet)
- shut down and restart wsl  (I'm on windows, running wsl2)
- Deleted the `/types/types.d.ts` file and deleted `components/ArticlesCell/` folder, and followed the steps again. 
  If I remove `ArticlesCell`, everything works fine. So I'm guessing there's something wrong in that file? 
  
  Here's the freshly generated content of Articles.ts: 

```
import type { ArticlesQuery } from 'types/graphql'
import type { CellSuccessProps, CellFailureProps } from '@redwoodjs/web'


export const QUERY = gql`
  query ArticlesQuery {
    articles {
      id
    }
  }
`

export const Loading = () => <div>Loading...</div>

export const Empty = () => <div>Empty</div>

export const Failure = ({ error }: CellFailureProps) => (
  <div style={{ color: 'red' }}>Error: {error.message}</div>
)

export const Success = ({ articles }: CellSuccessProps<ArticlesQuery>) => {
  return (
    <ul>
      {articles.map((item) => {
        return <li key={item.id}>{JSON.stringify(item)}</li>
      })}
    </ul>
  )
}  
 ```

It's weird, because ArticlesCell.ts imports `ArticlesQuery` from `types.d.ts`, and `types.d.ts` seems to be generated from `ArticlesQuery`? I'm guessing it's redwood-magic and typescript will stop complaining about the import when the type is actually generated. 

I would appreciate help to get this to work. 
Many thanks! :)

**Comment by MrEmanuel** [comment]
Also, I just downloaded everything today so I'm pretty sure I got everything up to date. 
My `package.json` files:
in web: 
```
{
  "name": "web",
  "version": "0.0.0",
  "private": true,
  "browserslist": {
    "development": [
      "last 1 version"
    ],
    "production": [
      "defaults",
      "not IE 11",
      "not IE_Mob 11"
    ]
  },
  "dependencies": {
    "@redwoodjs/forms": "1.5.1",
    "@redwoodjs/router": "1.5.1",
    "@redwoodjs/web": "1.5.1",
    "prop-types": "15.8.1",
    "react": "17.0.2",
    "react-dom": "17.0.2"
  }
}
```

In the root: 
```
{
  "private": true,
  "workspaces": {
    "packages": [
      "api",
      "web",
      "packages/*"
    ]
  },
  "devDependencies": {
    "@redwoodjs/core": "1.5.1"
  },
  "eslintConfig": {
    "extends": "@redwoodjs/eslint-config",
    "root": true
  },
  "engines": {
    "node": ">=14.19 <=16.x",
    "yarn": ">=1.15"
  },
  "prisma": {
    "seed": "yarn rw exec seed"
  },
  "packageManager": "yarn@3.2.1"
}

```

`yarn --version` 3.2.1 
`node --version` v16.13.1
`yarn rw check`  Success: no errors or warnings were detected (I added NODE_ENV to .env just now)
`yarn rw info`: 
```
  System:
    OS: Linux 4.19 Ubuntu 20.04.3 LTS (Focal Fossa)
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.13.1 - /tmp/xfs-2eafe6d6/node
    Yarn: 3.2.1 - /tmp/xfs-2eafe6d6/yarn
  npmPackages:
    @redwoodjs/core: 1.5.1 => 1.5.1
```

**Comment by MrEmanuel** [comment]
I noticed I have yarn 3. 
So I downgraded, made another project and tried creating a cell again. 

```
emanuel@Emanuel-Razer-Laptop:~/source/redwoodblog2$ yarn rw g cell Articles
yarn run v1.23.0-20220130.1630
$ /home/emanuel/source/redwoodblog2/node_modules/.bin/rw g cell Articles
(node:9360) ExperimentalWarning: stream/web is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
  ✔ Generating cell files...
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.mock.ts`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.test.tsx`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.stories.tsx`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.tsx`
  ⠹ Generating types ...

Error: Could not generate GraphQL type definitions (web)
AggregateError: GraphQL Document Validation failed with 1 errors;
  Error 0: GraphQLDocumentError: Cannot query field "articles" on type "Query".
    at /home/emanuel/source/redwoodblog2/web/src/components/ArticlesCell/ArticlesCell.tsx:2:3
    at Object.checkValidationErrors (/home/emanuel/source/redwoodblog2/node_modules/@graphql-tools/utils/index.js:1277:15)
  ✔ Generating cell files...
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.mock.ts`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.test.tsx`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.stories.tsx`
    ✔ Successfully wrote file `./web/src/components/ArticlesCell/ArticlesCell.tsx`
  ✔ Generating types ...
Done in 2.50s.
emanuel@Emanuel-Razer-Laptop:~/source/redwoodblog2$ yarn --version
**1.23.0-20220130.1630**
```
As you can see, I verified I have the correct version of yarn.

**Comment by MrEmanuel** [comment]
It turns out this is a pedagogical issue. 

On this page: https://redwoodjs.com/docs/tutorial/chapter2/cells 
After this command: `yarn rw g cell Articles`
I was expecting the app to work. I restarted the dev server and got the above mentioned error. Then I started searching high and low for what the issue was... Something wrong with the graphql-server? The type generations didn't seem to be working.. No `articles` anywhere.. Consulted stack overflow.. Did some google-fu...  Opened this issue. 

2 days later, reinstalled vscode, cleaned up my hosts file, removed lots of vscode addons, reinstalled addons, cleaned yarn cache, etc, etc... and now I tried to create a new redwood project a forth time. 

Suggestion: Right before or after telling me to run `yarn rw g cell Articles`, also let me know the app will break and we're about to manually fix it. My assumption was that the generated code would work. 

Letting the reader know when the app will break before running the command will hopefully save more people from going down a similar rabbit hole as I did. 


Thanks! :)

**Comment by jtoar** [comment]
Hey @MrEmanuel we're very sorry you had to go through all of this! 😔 Your persistence was incredible though, so I commend you. We've been needing better DX around this for some time now. See https://github.com/redwoodjs/redwood/issues/5538.

> My assumption was that the generated code would work

That assumption is correct. And the generated code should work—it's generated code at the beginning of the tutorial!

> I was expecting the app to work

One clarification here: the dev server does work doesn't it? It's just that you get an "gen" error about types right?

I'll work with the team to reprioritize this.

**Comment by MrEmanuel** [comment]
@jtoar Thanks for your reply. 
The dev server starts and runs fine, even if I got the error. I mentioned this in one of my bullet points. 

What everything boiled down to was that I wasn't comfortable to continue the tutorial with an active error. I didn't know it was part of the tutorial and that it was going to be fixed later. :) 

So, I managed to continue the tutorial and I'm well on my way to finish it. 
Thanks!
